### PR TITLE
Container image with installer script and dependencies

### PR DESCRIPTION
**Pull request by aishwaryabk**
Signed-off-by: Aishwarya Kamat <aishwarya.kamat@ibm.com>

### CODE REVIEW

File: images/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+WORKDIR $INSTALL_DIR
+
+RUN microdnf update && \
+    microdnf install -y yum findutils && \
+    mkdir -p $INSTALL_DIR
+
+RUN yum install -y openssh openssh-clients
+
+RUN curl -sL https://raw.githubusercontent.com/ocp-power-automation/openshift-install-power/master/openshift-install-powervs -o /usr/bin/openshift-install-powervs && \
```
Comment (yussufsh) [inline images/Dockerfile:17]: Why not COPY/ADD from current cloned directory?

File: images/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+
+ARG RELEASE_VER=4.6
```
Comment (yussufsh) [inline images/Dockerfile:4]: Please set this to the latest version.

We will have to write a release pipeline for deploying different versions of this Docker image. As ppc64le is also involved we can look at Travis jobs during `tag` events. @bpradipt

File: images/Dockerfile (hunk 1 above)
Comment (aishwaryabk) [reply images/Dockerfile]: Modified the Dockerfile

File: images/Dockerfile (hunk 2 above)
Comment (aishwaryabk) [reply images/Dockerfile]: Updated RELEASE_VER to 4.7

**Comment by yussufsh** [comment]
Thanks @aishwaryabk I will test this PR when I get to and let you know of any feedback.

**Comment by bpradipt** [comment]
@rpsene ptal.
This PR is to add support for running the install helper (bash wrapper to ocp4-upi-powervs) from a containerized deployment

**Comment by rpsene** [comment]
@aishwaryabk an end user will not care too much about building the image, they will just want to run it setting the required variables.  As it is today, there will be a lot of manual steps to be handled, with slow down the process of getting agile by using a container. 

You can get a base UBI image with all dependencies already installed from here https://quay.io/repository/powercloud/powervs-container-host?tab=tags. The you can set a file with all variables as parameter for the container (like the env file here https://github.com/rpsene/openshift-on-powervs-quick-deploy/blob/main/deploy.sh#L98-L103 and here https://github.com/rpsene/openshift-on-powervs-quick-deploy/blob/main/variables) and run the script to install it. When running exec, you need to ensure you keep a copy of the logs outside the container (leveraging the mount point you create with -v, for instance https://github.com/rpsene/openshift-on-powervs-quick-deploy/blob/main/scripts/run-terraform.sh#L36-L53).

If you apply this set of changes, anyone will be able to run what you have as part of anything, including using it from within another OCP cluster, as a pod, like https://github.com/ocp-power-automation/ocp-upi-powervs-pod.

Let me know if you have questions or want to meet; I have been using OCP deployments on PowerVS from containers for some time at IBM and RH.

**Comment by yussufsh** [comment]
@rpsene The problem she is facing that the script as an entrypoint when run from the host install directory does not use the binary executable (terraform, ibmcloud, oc, etc.) which are already downloaded in the image. The script does not find the executable in $PWD and downloads all of them again.

Did you come across such scenario before?

**Comment by yussufsh** [comment]
@aishwaryabk Please remove the documentation from this PR. Once the https://github.com/ocp-power-automation/openshift-install-power/pull/140 is merged I will remove the do-not-merge tag.

Once we have the dockerimage in the repo, we need Travis pipeline for creating and publishing the images as per release version. After that we can update the documentation on how to use the docker image.

**Comment by aishwaryabk** [comment]
> @aishwaryabk Please remove the documentation from this PR. Once the #140 is merged I will remove the do-not-merge tag.
> 
> Once we have the dockerimage in the repo, we need Travis pipeline for creating and publishing the images as per release version. After that we can update the documentation on how to use the docker image.

@yussufsh Updated the changes.

**Comment by lsmcfadden** [comment]
@aishwaryabk looks like you are making progress here.  Still targeting to wrap in this sprint?

**Review by yussufsh — APPROVED** [review]
lgtm

**Pull request merged**